### PR TITLE
(firestore): distributed counters examples

### PR DESCRIPTION
**Pull request by jiren**

### CODE REVIEW

File: firestore/distributed_counters.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "google/cloud/firestore"
+
+def create_counter project_id:, num_shards:
+  # project_id = "Your Google Cloud Project ID"
+  # num_shards = "Number of shards for distributed counter"
+
+  firestore = Google::Cloud::Firestore.new project_id: project_id
```
Comment (TheRoyalTnetennba) [inline firestore/distributed_counters.rb]: Can you include the client creation and the `require "google/cloud/firestore"` within the region tags

**Comment by IlyaFaer** [comment]
@TheRoyalTnetennba, could you take another look, please

File: firestore/distributed_counters.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  # [START fs_create_counter]
+  shards_ref = firestore.col "shards"
+
+  # Initialize each shard with count=0
+  num_shards.times do |i|
+    shards_ref.doc(i).set(count: 0)
+  end
+
+  # [END fs_create_counter]
+  puts "Distributed counter shards collection created."
```
Comment (TheRoyalTnetennba) [inline firestore/distributed_counters.rb]:
```suggestion
  # [START fs_create_counter]
  # project_id = "Your Google Cloud Project ID"
  # num_shards = "Number of shards for distributed counter"

  require "google/cloud/firestore"

  firestore = Google::Cloud::Firestore.new project_id: project_id

  shards_ref = firestore.col "shards"

  # Initialize each shard with count=0
  num_shards.times do |i|
    shards_ref.doc(i).set(count: 0)
  end

  puts "Distributed counter shards collection created."
  # [END fs_create_counter]
```

generally, snippets are meant to be copy-paste runnable, so both the requires and the client initialization should be included in the region tags

Comment (jiren) [reply firestore/distributed_counters.rb]: @TheRoyalTnetennba  updated code.

File: firestore/distributed_counters.rb [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+  # increment counter
+  shard_ref.update count: firestore.field_increment(1)
+
+  puts "Counter incremented."
+  # [END fs_increment_counter]
+end
+
+def get_count project_id:
+  # [START fs_increment_counter]
```
Comment (TheRoyalTnetennba) [inline firestore/distributed_counters.rb]:
```suggestion
  # [START fs_get_count
```

Comment (jiren) [reply firestore/distributed_counters.rb]: @TheRoyalTnetennba  done

**Comment by TheRoyalTnetennba** [comment]
do you need me to merge this?

**Comment by jiren** [comment]
> do you need me to merge this?

@TheRoyalTnetennba  Yes, Please merge it.

**Pull request merged**